File: Javascript/1751-1800/1752. Check if Array Is Sorted and Rotated.js
// https://leetcode.com/problems/check-if-array-is-sorted-and-rotated/

var check = function(nums) {
    let count = 0, size = nums.length;

    for(let index=1; index<size; index++){
        if(nums[index-1] > nums[index]) count++;
    }

    // Checks
    if(count == 0) return true; // Sorted Array
    else if(count > 1) return false;
    else if (nums[0] >= nums[size-1]) return true; // sorted and rotated

    return false;
};

File: Python/1751-1800/1752. Check if Array Is Sorted and Rotated.py
# https://leetcode.com/problems/check-if-array-is-sorted-and-rotated/

class Solution:
    def check(self, nums: List[int]) -> bool:
        count, size = 0, len(nums)

        for index in range (1, size):
            if nums[index-1] > nums[index]:
                count += 1

        # Checks
        if count == 0:
            return True # Sorted Array
        elif count > 1:
            return False
        elif nums[0] >= nums[size-1]:
            return True # sorted and rotated

        return False
